org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/impl/ChangeSerializer.java

```diff
@@ -80,17 +82,34 @@ else if (context instanceof Resource)
 			else if (context instanceof ResourceSet) {
 				throw new IllegalStateException("Not supported");
 			}
-			progress.split(1);
 		}
+		
+		// in the following effort calculation, the amount of resources is doubled for the sake of reflecting
+		//  the usually higher effort of indexing the contents of a resource compared to applying a modification
+		final int weightedPrepareAndApplyEffort = 2 * resources.size() + modifications.size();
+		
+		// here, the amount of resources is doubled for the sake of reflecting the still unknown amount of
+		//  related resources to be updated later on, too; the additional effort is a rough approximation of
+		//  the effort required to determine the (affected) related resources
+		final int weightedCreateTextChangesEffort = 2 * resources.size() + additionalEffortInEndRecordChanges;
+		
+		final SubMonitor monitorInAll = SubMonitor.convert(this.monitor,
+				weightedPrepareAndApplyEffort + weightedCreateTextChangesEffort);
+		final SubMonitor monitorPrepareAndApply = monitorInAll.split(weightedPrepareAndApplyEffort, SubMonitor.SUPPRESS_NONE);
+		monitorPrepareAndApply.beginTask("Preparing and applying file changes...", weightedPrepareAndApplyEffort);
```

Comment:
It is quite unusual to call `beginTask` on a SubMonitor. Instead of `split()` you might convert the monitor again:
```
final SubMonitor monitorPrepareAndApply = SubMonitor.convert(monitorInAll, "Preparing and applying file changes...", weightedPrepareAndApplyEffort);
```

Reply:
I prepare a test concerning the progress reporting, and I implemented it s.t. my expectations are met, but I'll have a look.